### PR TITLE
Rework/schedule management

### PR DESCRIPTION
**Pull request by v-stamenova**
The PR introduces a lot of changes on the scheduling process:
Run `sail artisan migrate` because a new table is introduced - `default_presentations`

For testing:
---
The new schedule flow
- [x] Create opening for the event
- [x] Create closing for the event
- [x] Give minutes that would be between the timeslots
---
After the steps above - before releasing the programme
- [x] Automatic schedule generation (from the schedule overview)
- [x] Manually assigning room and timeslot
- [x] Changing the assigned room and timeslot
- [x] Removing presentation from the schedule
- [x] Possibility of deleting the presentation (check presentation CRUD for that)
- [x] Resetting the schedule - you'll need to go through the steps mentioned in the new schedule flow again
- [x] Editing of opening - if the new ending time of the opening overlaps the timeslot of the first presentation, mod will be prompted to either reset all timeslots (without the opening and closing) or choose new ending time
- [x] Editing of closing - if the new starting time of the closing overlaps the timeslot of the last presentation, mod will be prompted to either reset all timeslots (without the opening and closing) or choose new starting time
- [x] Schedule is able to be released only after all presentations approved are scheduled
---
After releasing the programme
- [x] Removing presentation from the schedule
- [x] Not possible to fully delete presentations - only to remove it from the schedule but without removing it fully
- [x] Replacing a presentation in the schedule with presentation that is approved but not in the schedule
- [x] A presentation that is approved but not in the schedule can be manually added to the schedule

### CODE REVIEW

**Comment by TimKardol** [comment]
Found an Issue: when editing the first presentation (not the opening one) and changing the timeslot to a later one, it stays at the top of the list. It doesn't go to the bottom. See screenshot:
image

File: resources/views/moderator/schedule/presentations/remove-presentation-from-schedule.blade.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div>
+    <div class="font-semibold pb-2">
+        Remove presentation
+    </div>
+    <p class="text-md pb-1">If you want you cannot replace the presentation, but you want it off the schedule, you can
```
Comment (TimKardol) [inline resources/views/moderator/schedule/presentations/remove-presentation-from-schedule.blade.php:5]: Question: Is there a way to replace the presentation from this page? The grammar seems off.
```suggestion
    <p class="text-md pb-1">If you choose not to replace the presentation, you can just take it off the schedule.</p>
```

**Comment by TimKardol** [comment]
Issue: Found this when trying to replace a presentation that is already approved with another presentation.
image

File: resources/views/livewire/room-and-timeslot-selector.blade.php [hunk 2]
```diff
@@ -4,28 +4,33 @@
...
-                    class="mt-1 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-indigo-500 focus:border-indigo-500 block w-full p-2.5 dark:bg-gray-900 dark:border-gray-700 dark:placeholder-gray-400 dark:text-white dark:focus:ring-indigo-500 dark:focus:border-indigo-500">
+                    class="mt-1 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-crew-500 focus:border-crew-500 block w-full p-2.5 dark:bg-gray-900 dark:border-gray-700 dark:placeholder-gray-400 dark:text-white dark:focus:ring-crew-500 dark:focus:border-crew-500">
                 @if(is_null($presentation->room))
                     <option value="">Select a room...</option>
                 @endif
                 @foreach ($rooms as $room)
                     @if($presentation->room_id == $room->id)
-                        <option selected value="{{ $room->id }}">{{ $room->name }} (max participants: {{$room->max_participants}})</option>
+                        <option selected value="{{ $room->id }}">{{ $room->name }} (max
+                                                                                   participants: {{$room->max_participants}}
```
Comment (TimKardol) [inline resources/views/livewire/room-and-timeslot-selector.blade.php:16]: Indent is weird here. Could be a single line right?

File: resources/views/livewire/room-and-timeslot-selector.blade.php [hunk 3]
```diff
@@ -4,28 +4,33 @@
...
+                                                                                   )
+                        </option>
                     @else
-                        <option value="{{ $room->id }}">{{ $room->name }}  (max participants: {{$room->max_participants}})</option>
+                        <option value="{{ $room->id }}">{{ $room->name }} (max participants: {{$room->max_participants}}
+                                                                          )
```
Comment (TimKardol) [inline resources/views/livewire/room-and-timeslot-selector.blade.php:21]: Same here

File: resources/views/moderator/requests/presentations.blade.php [hunk 4]
```diff
@@ -1,55 +1,71 @@
...
-                            <div class="mt-2 ml-4">
-                                <h3 class="text-gray-900 dark:text-white leading-6 font-semibold text-base">All presentations</h3>
+            <div class="px-8 py-6 max-w-7xl mx-auto">
+                <div class="max-w-none mx-auto">
+                    <div class="rounded-lg bg-white dark:bg-slate-700 overflow-hidden">
+                        <div class="px-6 py-5 bg-white dark:bg-slate-700 border-b dark:border-gray-500">
+                            <div class="flex-nowrap justify-between items-center flex -mt-2 -ml-4">
+                                <div class="mt-2 ml-4">
+                                    <h3 class="text-gray-900 dark:text-white leading-6 font-semibold text-base">All
+                                                                                                                presentations</h3>
```
Comment (TimKardol) [inline resources/views/moderator/requests/presentations.blade.php:17]: Weird indentation again

File: resources/views/moderator/schedule/default-presentations/edit-default-presentation-form.blade.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    <x-dialog-modal wire:model="confirmationTimeslotRegeneration">
+        <x-slot name="title">
+            {{ __('New ending time') }}
+        </x-slot>
+
+        <x-slot name="content">
+            @if($presentation->type == 'opening')
+                <h3 class="font-bold text-red-600">{{ __('WARNING: You set a new time for ending of the opening presentation for a later time
+                than initially stated. This means that timeslots will start to overlap. In order to proceed you need to regenerate the timeslots
+                meaning that all scheduled presentations would lose their scheduling and you need to reschedule them again. Do you want to proceed?') }}
```
Comment (TimKardol) [inline resources/views/moderator/schedule/default-presentations/edit-default-presentation-form.blade.php:56]:
```suggestion
                <h3 class="font-bold text-red-600">{{ __('WARNING: By extending the end time of the opening presentation, timeslots will overlap. If you proceed, all presentations will need to be rescheduled. Do you want to continue? ') }}
```

File: resources/views/moderator/schedule/default-presentations/edit-default-presentation-form.blade.php [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                    @else
+                        <h3 class="font-bold text-red-600">{{ __('WARNING: You set a new time for starting of the closing presentation for an earlier time
+                than initially stated. This means that timeslots will start to overlap. In order to proceed you need to regenerate the timeslots
+                meaning that all scheduled presentations would lose their scheduling and you need to reschedule them again. Do you want to proceed?') }}
```
Comment (TimKardol) [inline resources/views/moderator/schedule/default-presentations/edit-default-presentation-form.blade.php:60]:
```suggestion
                        <h3 class="font-bold text-red-600">{{ __('WARNING: You've moved the start time of the closing presentation to an earlier slot, causing an overlap in timeslots. To continue, all presentations will need to be rescheduled. Do you want to continue?') }}
```

File: resources/views/moderator/schedule/default-presentations/edit.blade.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        </h1>
+        <div class="pt-5">
+            <div
+                class="mt-5 gap-6 text-gray-900 dark:text-gray-200 px-4 py-5 sm:p-6 bg-white dark:bg-gray-800 shadow sm:rounded-md">
+                <div class="pt-1 p-2">
+                    <div class="pr-5">
+                        <p class="text-md text-gray-900 dark:text-white">
+                            @if($event == 'opening')
+                                Create the opening presentation, give more details about the event that will be
+                                kicking off the conference day.
```
Comment (TimKardol) [inline resources/views/moderator/schedule/default-presentations/edit.blade.php:14]:
```suggestion
                                Create the opening presentation. You can add more details about the opening presentation.
```

File: resources/views/moderator/schedule/default-presentations/edit.blade.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                            @elseif($event == 'closing')
+                                Create the closing presentation, give more details about the event that will end the
+                                conference day on a high note.
```
Comment (TimKardol) [inline resources/views/moderator/schedule/default-presentations/edit.blade.php:17]:
```suggestion
                                Create the closing presentation. You can add more details about the closing presentation.
```

File: resources/views/moderator/schedule/presentations/index.blade.php [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                                                               scheduling</h2>
+
+            <div class="px-8 py-6 max-w-7xl mx-auto">
+                <div class="max-w-none mx-auto">
+                    <div class="rounded-lg bg-white dark:bg-slate-700 overflow-hidden">
+                        <div class="px-6 py-5 bg-white dark:bg-slate-700 border-b dark:border-gray-500">
+                            <div class="flex-nowrap justify-between items-center flex -mt-2 -ml-4">
+                                <div class="mt-2 ml-4">
+                                    <h3 class="text-gray-900 dark:text-white leading-6 font-semibold text-base">All
+                                                                                                                presentations</h3>
```
Comment (TimKardol) [inline resources/views/moderator/schedule/presentations/index.blade.php:17]: Weird indent

File: resources/views/moderator/schedule/timeslots/add-timeslot-padding.blade.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    </x-button>
+
+    <x-dialog-modal wire:model="isModalOpen">
+        <x-slot name="title">
+            {{ __('Add time between timeslots') }}
+        </x-slot>
+
+        <x-slot name="content">
+            <h3 class="font-bold text-gray-600 dark:text-gray-200">{{ __('In order to make it possible for participants to change rooms for presentations, they need to have
+                enough time to calmly navigate through the building. The default minutes for this is 10 minutes. Would you like to change it?') }}</h3>
```
Comment (TimKardol) [inline resources/views/moderator/schedule/timeslots/add-timeslot-padding.blade.php:14]:
```suggestion
                enough time to calmly navigate through the building. The default changeover time is 10 minutes. Would you like to change it?') }}</h3>
```

**Review by TimKardol — CHANGES_REQUESTED** [review]

**Comment by v-stamenova** [comment]
> Found an Issue: when editing the first presentation (not the opening one) and changing the timeslot to a later one, it stays at the top of the list. It doesn't go to the bottom. See screenshot: image

> Issue: Found this when trying to replace a presentation that is already approved with another presentation. image

Fixed

**Review by TimKardol — APPROVED** [review]

**Pull request merged**